database/Table_Insert.py

```diff
@@ -116,4 +124,7 @@ def prepare_data_for_insertion(schema, data):
         if ip_recorded[0][0] > 0:
             pass
         else:
-            insert_data(table_name, insert_list, None , has_id)
\ No newline at end of file
+            insert_data(table_name, insert_list, None , has_id)
+
+    connection.commit()
```

Comment:
it would probably be good practice to write this whole block in a try/finally to ensure we always close the connection.  report server has this same problem going on - so who am I to talk.  :-(